sydent/http/servlets/msisdnservlet.py

```diff
@@ -150,18 +166,19 @@ def do_validate_request(self, args):
             }
 
         try:
-            resp = self.sydent.validators.msisdn.validateSessionWithToken(sid, clientSecret, tokenString)
+            return self.sydent.validators.msisdn.validateSessionWithToken(sid, clientSecret, tokenString)
         except IncorrectClientSecretException:
-            return {'success': False, 'errcode': 'M_INCORRECT_CLIENT_SECRET',
+            return {'success': False, 'errcode': 'M_INVALID_PARAM',
                     'error': "Client secret does not match the one given when requesting the token"}
         except SessionExpiredException:
             return {'success': False, 'errcode': 'M_SESSION_EXPIRED',
                     'error': "This validation session has expired: call requestToken again"}
-
-        if not resp:
-            resp = {'success': False}
-
-        return resp
+        except InvalidSessionIdException:
+            return {'success': False, 'errcode': 'M_INVALID_PARAM',
+                    'error': "The token doesn't match"}
+        except IncorrectSessionTokenException:
+            return {'success': False, 'errcode': 'M_NO_VALID_SESSION',
+                    'error': "No session could be found with this sid"}
```

Comment:
These don't look like python3 related changes?

Reply:
It's not, but it looked like a quick win I could fix along the way. lmk if it shouldn't be in this PR.

Reply:
In general I'd prefer functional changes to be outside, simply because then the PR is just a matter of making sure all changes are no-op rather than worrying about if they're correct

Reply:
kk, I'll do that then

Reply:
IRL: it's actually probably fine in this case, next time let's make another PR and base the extra work on top of that.